hbase-client/src/main/java/org/apache/hadoop/hbase/shaded/protobuf/RequestConverter.java

```diff
@@ -346,10 +346,9 @@ public static RegionAction.Builder getRegionActionBuilderWithRegion(
   public static ScanRequest buildScanRequest(byte[] regionName, Scan scan, int numberOfRows,
     boolean closeScanner) throws IOException {
     ScanRequest.Builder builder = ScanRequest.newBuilder();
-    RegionSpecifier region = buildRegionSpecifier(RegionSpecifierType.REGION_NAME, regionName);
+    builder.setRegion(buildRegionSpecifier(RegionSpecifierType.REGION_NAME, regionName));
```

Comment:
Because the way I obtain the table name is through the region name. However, for a scan query, the region name will only be passed in the first request, and subsequent requests will no longer pass the region name but the scannerId. Therefore, this class needs to be modified, otherwise the table metrics for the scan request cannot be obtained.

Reply:
Thanks for explaining.

I'm not sure we should modify the RPC protocol just for the sake of these client side metrics. It might be better to use HBaseRpcController to shuttle the table name into the `onCallFinished` method (where `updateRpc` is called). As an example, you could modify AsyncScanSingleRegionRpcRetryingCaller.call method:

```diff
- resetController(controller, callTimeoutNs, priority);
+ resetController(controller, callTimeoutNs, priority, loc.getRegion().getTable());
ScanRequest req = RequestConverter.buildScanRequest(scannerId, scan.getCaching(), false,
      nextCallSeq, scan.isScanMetricsEnabled(), false, scan.getLimit());
final Context context = Context.current();
stub.scan(controller, req, resp -> {
      try (Scope ignored = context.makeCurrent()) {
        onComplete(controller, resp);
      }
});
```

This requires a bit more changes, but at least it keeps our RPC protocol unchanged.

You'll need to add a setTableName and getTableName to HBaseRpcController and DelegatingHBaseRpcController. Then update that `ConnectionUtils.resetController` method (and callers), as well as RegionServerCallable once on branch-2.

@Apache9 if you have time, do you agree with this advice?

Reply:
@Apache9 @bbeaudreault hi, could you provide me with more assistance so that I can complete this PR? Thank you very much.

Reply:
Is there something else you’d need? I’d go down the path I described, which I feel pretty good about. Just duo often has nice ideas so I figured I’d ask, but he’s probably busy.

Reply:
@bbeaudreault Thank you for your reply! Okay, I will make modifications as you described.

Reply:
@bbeaudreault hi, I have made the modifications as per your request. Could you please review the code and provide some suggestions? I would greatly appreciate it!